app/controllers/PropertyLinkingController.scala

```diff
@@ -193,30 +120,36 @@ class PropertyLinkingController @Inject()(
     }.map(_.flatten)
   }
 
-  def clientProperty(authorisationId: Long, clientOrgId: Long, agentOrgId: Long) = authenticated { implicit request =>
-    propertyLinksConnector.get(authorisationId) flatMap {
-      case Some(authorisation) if authorisedFor(authorisation, clientOrgId, agentOrgId) => toClientProperty(authorisation) map { p => Ok(Json.toJson(p)) }
-      case _ => NotFound
-    }
+  def getMyOrganisationsAssessments(submissionId: String) = authenticated { implicit request =>
+    implicit val cache = Memoize[Long, Future[Option[GroupAccount]]]()
+
+    assessmentService.getMyOrganisationsAssessments(submissionId).fold(Ok(Json.toJson(submissionId))) {authorisation =>
+      println(Json.toJson(authorisation))
+      Ok(Json.toJson(authorisation))}
   }
 
-  private def authorisedFor(authorisation: PropertiesView, clientOrgId: Long, agentOrgId: Long) = {
-    authorisation.authorisationOwnerOrganisationId == clientOrgId && authorisation.parties.exists(_.authorisedPartyOrganisationId == agentOrgId)
+  def getClientsAssessments(submissionId: String) = authenticated { implicit request =>
+    implicit val cache = Memoize[Long, Future[Option[GroupAccount]]]()
+
+    assessmentService.getClientsAssessments(submissionId).fold(Ok(Json.toJson(submissionId))) {authorisation =>
```

Comment:
the fold here should not throw a Ok on both.

Reply:
This was something I was going to raise, but obviously forgot. Do we want to return a NotFound in the left fold?? Had some trouble implementing this and could do with some help...